maas/resource_maas_boot_source_selection.go

```diff
@@ -0,0 +1,227 @@
+package maas
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMAASBootSourceSelection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a resource to manage a MAAS boot source selection.",
+		CreateContext: resourceBootSourceSelectionCreate,
+		ReadContext:   resourceBootSourceSelectionRead,
+		UpdateContext: resourceBootSourceSelectionUpdate,
+		DeleteContext: resourceBootSourceSelectionDelete,
+
```

Comment:
We are missing an import function. Content: https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/import